lib/src/main/java/org/exoplatform/social/addons/storage/entity/SpaceEntity.java

```diff
@@ -0,0 +1,290 @@
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.exoplatform.social.addons.storage.entity;
+
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Set;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.NamedQueries;
+import javax.persistence.NamedQuery;
+import javax.persistence.OneToMany;
+import javax.persistence.SequenceGenerator;
+import javax.persistence.Table;
+
+import org.exoplatform.commons.api.persistence.ExoEntity;
+import org.exoplatform.social.addons.storage.entity.SpaceMember.Status;
+import org.exoplatform.social.core.space.model.Space;
+
+@Entity
+@ExoEntity
+@Table(name = "SOC_SPACES")
+@NamedQueries({
+    @NamedQuery(name = "SpaceEntity.getLastSpaces", query = "SELECT sp FROM SpaceEntity sp ORDER BY sp.createdTime DESC"),
+    @NamedQuery(name = "SpaceEntity.getVisitedSpaces", query = "SELECT sp FROM SpaceEntity sp INNER JOIN sp.members as mem WHERE mem.userId = :userId AND mem.status = :status AND sp.app like :app ORDER BY mem.visited DESC, sp.prettyName ASC"),
+    @NamedQuery(name = "SpaceEntity.getLastAccessedSpace", query = "SELECT sp FROM SpaceEntity sp INNER JOIN sp.members as mem WHERE mem.userId = :userId AND mem.status = :status AND sp.app like :app ORDER BY mem.lastAccess DESC"),
+    @NamedQuery(name = "SpaceEntity.getSpaceByGroupId", query = "SELECT sp FROM SpaceEntity sp WHERE sp.groupId = :groupId"),
+    @NamedQuery(name = "SpaceEntity.getSpaceByPrettyName", query = "SELECT sp FROM SpaceEntity sp WHERE sp.prettyName = :prettyName"),
+    @NamedQuery(name = "SpaceEntity.getSpaceByDisplayName", query = "SELECT sp FROM SpaceEntity sp WHERE sp.displayName = :displayName"),
+    @NamedQuery(name = "SpaceEntity.getSpaceByURL", query = "SELECT sp FROM SpaceEntity sp WHERE sp.url = :url") })
+public class SpaceEntity implements Serializable {
+
+  private static final long serialVersionUID = 3223615477747436986L;
+
+  @Id
+  @SequenceGenerator(name = "SEQ_SOC_SPACES_ID", sequenceName = "SEQ_SOC_SPACES_ID")
+  @GeneratedValue(strategy = GenerationType.AUTO, generator = "SEQ_SOC_SPACES_ID")
+  @Column(name = "SPACE_ID")
+  private Long              id;
+
+  @OneToMany(fetch = FetchType.LAZY, mappedBy = "space", cascade = CascadeType.ALL, orphanRemoval = true)
+  private Set<SpaceMember> members          = new HashSet<>();
+
+  /**
+   * The list of applications with portlet Id, application name, and its state
+   * (installed, activated, deactivated).
+   */
+  @Column(name = "APP", length = 2000)
+  private String            app;
```

Comment:
This property has been modelized as a String in JCR because of the limitations of the JCR but we should thing relational database now. I think it has to be redesigned.